MatrixSDK/ThreePidAdd/MX3PidAddManager.m

```diff
@@ -94,6 +138,37 @@ - (MX3PidAddSession*)startAddEmailSessionWithEmail:(NSString*)email
 - (void)tryFinaliseAddEmailSession:(MX3PidAddSession*)threePidAddSession
                            success:(void (^)(void))success
                            failure:(void (^)(NSError * _Nonnull))failure
+{
+    [self tryFinaliseAddEmailSession:threePidAddSession authParams:nil success:success failure:failure];
+}
+
+- (void)tryFinaliseAddEmailSession:(MX3PidAddSession*)threePidAddSession
+                      withPassword:(nullable NSString*)password
+                           success:(void (^)(void))success
+                           failure:(void (^)(NSError * _Nonnull))failure
+{
+    // Make a first request to start user-interactive authentication
+    MXWeakify(self);
+    [self tryFinaliseAddEmailSession:threePidAddSession authParams:nil success:success failure:^(NSError * _Nonnull error) {
+        MXStrongifyAndReturnIfNil(self);
+
+        NSDictionary *authParams = [self authParamsFromError:error andPassword:password];
```

Comment:
No specific error catching here, it can lead to an infinite loop.

Reply:
`[self authParamsFromError: andPassword:]` that can return nil will ensure there is no infinite loop.